tmt/base.py

```diff
@@ -1968,7 +1969,7 @@ def __init__(self,
                  *,
                  path: Path = Path.cwd(),
                  tree: Optional[fmf.Tree] = None,
-                 context: Optional[tmt.utils.FmfContextType] = None,
+                 fmf_context: Optional[tmt.utils.FmfContextType] = None,
```

Comment:
The pull request is marked as `no functional change` but this `tmt.Tree` parameter rename actually breaks users which are using custom python scripts with `Tree` and `context`. My guess is that @t184256 or @Athwale are using this. But anyway, even if not, perhaps, could we keep the `context` parameter here unchanged? I don't think that exactly here there would be a big confusion about what context should be provided when calling `tmt.Tree()`.

Reply:
I agree, `no functional change` is incorrect, I didn't realize someone might see the change outside of tmt code base.

> But anyway, even if not, perhaps, could we keep the context parameter here unchanged?

We could, but I for one don't think we *should*. The confusion is exactly my motivation, I for one **always** had to wonder what `context` or `self.context` meant, which one of them it was. Sticking to `context` here (and only here) would leave the exception in code - everywhere else but here it would be called `fmf_context`, and as a human, I would have to remember `context` here is not just any context, but *fmf* context - why not make it clear for people who are not nearly familiar with the code? I don't believe this place is special enough to justify an exception, for me, `fmf_` prefix improves clarity and helps close the window for guessing.

Reply:
Ok, thanks for the detailed explanation. Convinced me. Let's keep it as you proposed.